Challenge1/src/agent.spec.ts

```diff
@@ -1,74 +1,108 @@
-//   ERC20_TRANSFER_EVENT,
-//   TETHER_ADDRESS,
-//   TETHER_DECIMALS,
-// } from "./agent";
-
-// describe("high tether transfer agent", () => {
-//   let handleTransaction: HandleTransaction;
-//   const mockTxEvent = createTransactionEvent({} as any);
-
-//   beforeAll(() => {
-//     handleTransaction = agent.handleTransaction;
-//   });
-
-//   describe("handleTransaction", () => {
-//     it("returns empty findings if there are no Tether transfers", async () => {
-//       mockTxEvent.filterLog = jest.fn().mockReturnValue([]);
-
-//       const findings = await handleTransaction(mockTxEvent);
-
-//       expect(findings).toStrictEqual([]);
-//       expect(mockTxEvent.filterLog).toHaveBeenCalledTimes(1);
-//       expect(mockTxEvent.filterLog).toHaveBeenCalledWith(
-//         ERC20_TRANSFER_EVENT,
-//         TETHER_ADDRESS
-//       );
-//     });
-
-//     it("returns a finding if there is a Tether transfer over 10,000", async () => {
-//       const mockTetherTransferEvent = {
-//         args: {
-//           from: "0xabc",
-//           to: "0xdef",
-//           value: ethers.BigNumber.from("20000000000"), //20k with 6 decimals
-//         },
-//       };
-//       mockTxEvent.filterLog = jest
-//         .fn()
-//         .mockReturnValue([mockTetherTransferEvent]);
-
-//       const findings = await handleTransaction(mockTxEvent);
-
-//       const normalizedValue = mockTetherTransferEvent.args.value.div(
-//         10 ** TETHER_DECIMALS
-//       );
-//       expect(findings).toStrictEqual([
-//         Finding.fromObject({
-//           name: "High Tether Transfer",
-//           description: `High amount of USDT transferred: ${normalizedValue}`,
-//           alertId: "FORTA-1",
-//           severity: FindingSeverity.Low,
-//           type: FindingType.Info,
-//           metadata: {
-//             to: mockTetherTransferEvent.args.to,
-//             from: mockTetherTransferEvent.args.from,
-//           },
-//         }),
-//       ]);
-//       expect(mockTxEvent.filterLog).toHaveBeenCalledTimes(1);
-//       expect(mockTxEvent.filterLog).toHaveBeenCalledWith(
-//         ERC20_TRANSFER_EVENT,
-//         TETHER_ADDRESS
-//       );
-//     });
-//   });
-// });
+import { Finding, HandleTransaction, FindingSeverity, FindingType, TransactionEvent } from "forta-agent";
+import agent from "./agent";
+
+let handleTransaction: HandleTransaction;
+const mockNetherMindAddress = "0x88dC3a2284FA62e0027d6D6B1fCfDd2141a143b8";
+const mockFortaRegistryAddress = "0x61447385B019187daa48e91c55c02AF1F1f3F863";
+
+beforeAll(() => {
+  handleTransaction = agent.handleTransaction;
+});
+
+describe("returns empty findings if transaction is not from Nethermind address", () => {
+  it("should return empty findings", async () => {
+    const mockTxEvent = {
+      from: "0x1234567890123456789012345678901234567890",
+      to: mockFortaRegistryAddress,
+      filterFunction: jest.fn().mockReturnValue([]),
+    } as unknown as TransactionEvent;
```

Comment:
I would recommend you look into the `TestTransactionEvent` from the `forta-agent-tools` ([here](https://github.com/NethermindEth/general-agents-module?tab=readme-ov-file#testtransactionevent)). When the team builds TS bots, we usually opt for the `forta-agent-tools` library as a default. It also includes a lot of added functionality that has been added as various needs arise. 🙂